File: tests/e2e/nodejs_client/test/health.test.js
/**
 * E2E: health endpoints against a live bqemulator container.
 *
 * Run with: BQEMU_REST_URL=http://localhost:9050 npm test
 */

const { describe, it } = require("node:test");
const assert = require("node:assert/strict");

const REST_URL = process.env.BQEMU_REST_URL || "http://localhost:9050";

describe("bqemulator health (Node.js)", () => {
  it("GET /healthz returns ok", async () => {
    const res = await fetch(`${REST_URL}/healthz`);
    assert.equal(res.status, 200);
    const body = await res.json();
    assert.equal(body.status, "ok");
  });

  it("GET /readyz returns ok", async () => {
    const res = await fetch(`${REST_URL}/readyz`);
    assert.equal(res.status, 200);
    const body = await res.json();
    assert.equal(body.status, "ok");
  });
});

File: tests/e2e/java_client/src/test/java/com/example/HealthTest.java
package com.example;

import org.junit.jupiter.api.Test;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import static org.junit.jupiter.api.Assertions.*;

/**
 * E2E: health endpoints against a live bqemulator container.
 *
 * Run with: BQEMU_REST_URL=http://localhost:9050 mvn test
 */
class HealthTest {
    private static final String REST_URL =
        System.getenv("BQEMU_REST_URL") != null
            ? System.getenv("BQEMU_REST_URL")
            : "http://localhost:9050";

    @Test
    void healthzReturnsOk() throws Exception {
        var client = HttpClient.newHttpClient();
        var request = HttpRequest.newBuilder()
            .uri(URI.create(REST_URL + "/healthz"))
            .GET()
            .build();
        var response = client.send(request, HttpResponse.BodyHandlers.ofString());
        assertEquals(200, response.statusCode());
        assertTrue(response.body().contains("\"ok\""));
    }

    @Test
    void readyzReturnsOk() throws Exception {
        var client = HttpClient.newHttpClient();
        var request = HttpRequest.newBuilder()
            .uri(URI.create(REST_URL + "/readyz"))
            .GET()
            .build();
        var response = client.send(request, HttpResponse.BodyHandlers.ofString());
        assertEquals(200, response.statusCode());
    }
}
